src/pks/flow/constitutive_relations/sources/pipe_drain_evaluator.cc

```diff
@@ -0,0 +1,360 @@
+/* -*-  mode: c++; indent-tabs-mode: nil -*- */
+
+/*
+  Evaluator for water drainage through a pipe network.
+  This depends on:
+  1) flow depth above surface elevation (surface_depth_key_)
+  2) hydraulic head in the pipe network (pressure_head_key_)
+  3) pipe drain length that is determined by the bathymetry difference
+
+  Authors: Giacomo Capodaglio (gcapodaglio@lanl.gov)
+           Naren Vohra (vohra@lanl.gov)
+*/
+
+#include "pipe_drain_evaluator.hh"
+#include "Geometry.hh"
+
+namespace Amanzi {
+namespace Flow {
+
+
+PipeDrainEvaluator::PipeDrainEvaluator(Teuchos::ParameterList& plist)
+  : EvaluatorSecondaryMonotypeCV(plist)
+{
+  manhole_radius_ = plist_.get<double>("manhole radius", 0.24);
+  energ_loss_coeff_weir_ = plist.get<double>("energy losses coeff weir", 0.54);
+  energ_loss_coeff_subweir_ = plist.get<double>("energy losses coeff submerged weir", 0.056);
+  energ_loss_coeff_orifice_ = plist.get<double>("energy losses coeff orifice", 0.167);
+
+  sw_domain_name_ = plist.get<std::string>("surface domain name", "surface");
```

Comment:
Can one of these domains be inferred from the key being evaluated?  E.g, if you are evaluating "surface-pipe_drainage" then prefer to use:

```sw_domain_name_ = Keys::getDomain(my_keys_.front().first);```

Reply:
Thank you for your comment. 

The domain is obtained using "Keys::getDomain(my_keys_.front().first)" to identify if we are in the surface or pipe network domain later in the code (line 54) when using the pipe_drain_evaluator. 

Hence we need to set the default name of sw_ or pipe_ domain before we use "Keys::getDomain(my_keys_.front().first)" to obtain the current domain.